### PR TITLE
Fix Several Bugs

### PR DESCRIPTION
**Pull request by NeilKleistGao**

### CODE REVIEW

File: shared/src/test/diff/codegen/Mixin.mls [hunk 1]
```diff
@@ -498,14 +514,120 @@ class C(x: int) extends BB
...
 
+:js
+mixin Fooo(x: int)
+//│ mixin Fooo()
+//│ // Prelude
+//│ let typing_unit14 = {
+//│   cache: {},
+//│   Fooo(base) {
+//│     if (base === undefined) {
+//│       return (class Fooo {
```
Comment (LPTK) [inline shared/src/test/diff/codegen/Mixin.mls]: It doesn't seem like the best idea to duplicate the code of the entire mixin. There could be a lot of code there, producing a lot of bloat. Why not just pass `Object` when `base` is undefined? The rest of the class definition doesn't need to differ.

File: shared/src/test/diff/codegen/Super.mls [hunk 2]
```diff
@@ -94,10 +94,9 @@ module Test0 extends Foo0, Foo1
 //│     = [ 1, 0 ]
 
 
-// FIXME some errors are currently swallowed!
-
 :ShowRepl
 :js
+:ge
```
Comment (LPTK) [inline shared/src/test/diff/codegen/Super.mls]:
```suggestion
// FIXME prevent misuse of `super` using the type system
```

File: shared/src/test/diff/nu/ModuleParameters.mls [hunk 3]
```diff
@@ -10,21 +10,15 @@ module A(x: int) { fun y = x }
...
-//│ Code generation encountered an error:
-//│   unresolved symbol x
 
-:re
 A.x
 //│ int
 //│ res
-//│ Runtime error:
-//│   ReferenceError: A is not defined
+//│     = undefined
```
Comment (LPTK) [inline shared/src/test/diff/nu/ModuleParameters.mls:17]: Why does it return `undefined`? Shouldn't it just access `this.#x`?

**Review by LPTK — COMMENTED** [review]
Nice progress, though we're still not quite there yet!

File: shared/src/test/diff/nu/ModuleParameters.mls (hunk 3 above)
Comment (NeilKleistGao) [reply shared/src/test/diff/nu/ModuleParameters.mls:17]: Because we have no parameter for `module A(x: int) { fun y = x }`.

Comment (LPTK) [reply shared/src/test/diff/nu/ModuleParameters.mls:17]: Oh never mind this one is okay. But there are other unexplained `undefined`s in this PR.

File: shared/src/test/diff/nu/ParamOverride.mls [hunk 4]
```diff
@@ -49,7 +49,7 @@ class Test0 extends Base1(0), Derived1(1) // TODO
 let t = Test0()
 //│ let t: Test0
 //│ t
-//│   = Test0 { n: 1 }
+//│   = Test0 { n: 0 }
```
Comment (LPTK) [inline shared/src/test/diff/nu/ParamOverride.mls]: Shouldn't this be `1`? I thought `Derived` would override it.

File: shared/src/test/diff/nu/ParamOverride.mls [hunk 5]
```diff
@@ -68,7 +68,7 @@ t.foo
 //│ ╙──      	 ^^^^
 //│ error
 //│ res
-//│     = [ 1, undefined ]
+//│     = [ 0, undefined ]
```
Comment (LPTK) [inline shared/src/test/diff/nu/ParamOverride.mls]: Why `undefined`?

File: shared/src/test/diff/codegen/Mixin.mls (hunk 1 above)
Comment (NeilKleistGao) [reply shared/src/test/diff/codegen/Mixin.mls]: Good idea.

File: shared/src/test/diff/nu/ModuleParameters.mls (hunk 3 above)
Comment (NeilKleistGao) [reply shared/src/test/diff/nu/ModuleParameters.mls:17]: Accessing `this.#x` is still under progress

**Comment by LPTK** [comment]
This is incorrect:

```fs
:js
mixin Derived1(n: int) {
  fun foo = [n, this.n, super.n]
}
//│ mixin Derived1() {
//│   super: {n: 'n}
//│   this: {n: 'n0}
//│   fun foo: (int, 'n0, 'n,)
//│ }
//│ // Prelude
//│ let typing_unit4 = {
//│   cache: {},
//│   Derived1(base) {
//│     if (base === undefined) {
//│       return (class Derived1 {
//│         #n;
//│         constructor(n) {
//│           this.n = n;
//│           this.#n = n;
//│         }
//│         get foo() {
//│           const n = this.n;
//│           const self = this;
//│           return ([
//│             n,
//│             self.n,
//│             super.n
//│           ]);
//│         }
//│       });
//│     } else {
//│       return (class Derived1 extends base {
//│         #n;
//│         constructor(n, ...rest) {
//│           super(...rest);
//│           this.n = n;
//│           this.#n = n;
//│         }
//│         get foo() {
//│           const n = this.n;
//│           const self = this;
//│           return ([
//│             n,
//│             self.n,
//│             super.n
//│           ]);
//│         }
//│       });
//│     }
//│   }
//│ };
//│ globalThis.Derived1 = typing_unit4.Derived1;
//│ // End of generated code

class Test0 extends Base1(0), Derived1(1) // TODO
//│ ╔══[ERROR] Unsupported parent specification
//│ ║  l.40: 	class Test0 extends Base1(0), Derived1(1) // TODO
//│ ╙──      	                    ^^^^^^^^
//│ ╔══[ERROR] Unsupported parent specification
//│ ║  l.40: 	class Test0 extends Base1(0), Derived1(1) // TODO
//│ ╙──      	                              ^^^^^^^^^^^
//│ class Test0()

let t = Test0()
//│ let t: Test0
//│ t
//│   = Test0 { n: 0 }

t.foo
//│ ╔══[ERROR] class `Test0` does not contain member `foo`
//│ ║  l.56: 	t.foo
//│ ╙──      	 ^^^^
//│ error
//│ res
//│     = [ 0, 0, undefined ]
```

It should return `[1, 1, 0]`.

**Comment by LPTK** [comment]
So, I'd expect the class to look like this instead:

```js
       return (class Derived1 extends base {
         #n;
         get n() { return #n };
         constructor(n, ...rest) {
           super(...rest);
           this.#n = n;
         }
         get foo() {
           const n = this.#n;
           const self = this;
           return ([
             n,
             self.n,
             super.n
           ]);
         }
       });
```

Notice:
 - we don't need to register `n` as a field; it's now a getter (can be overridden on further mixins)
 - we access `this.#n` when referring directly to the naked `n`

**Comment by LPTK** [comment]
Here's a self-contained example of error that gets swallowed:

```fs
:NewParser
:NewDefs

:e
:ge
mixin Base {
  fun x = y
}
//│ ╔══[ERROR] identifier not found: y
//│ ║  l.8: 	  fun x = y
//│ ╙──     	          ^
//│ mixin Base() {
//│   fun x: error
//│ }
//│ Code generation encountered an error:
//│   unresolved symbol y


:ShowRepl
module Test extends Base
//│ module Test() {
//│   fun x: error
//│ }
//│ ┌ Block at WrapState.mls:21
//│ ├─┬ Prelude
//│ │ ├── Code
//│ │ │   let res;
//│ │ │   let typing_unit1 = {
//│ │ │     cache: {},
//│ │ │     get Test() {
//│ │ │       if (this.cache.Test === undefined) {
//│ │ │         class Test extends Base(Object) {
//│ │ │           constructor() {
//│ │ │             super();
//│ │ │           }
//│ │ │         }
//│ │ │         this.cache.Test = new Test();
//│ │ │         this.cache.Test["class"] = Test;
//│ │ │       }
//│ │ │       return this.cache.Test;
//│ │ │     }
//│ │ │   };
//│ │ │   globalThis.Test = typing_unit1.Test;
//│ │ └── Reply
//│ │     Uncaught ReferenceError: Base is not defined
//│ │         at get Test [as Test] (REPL2:1:83)
//│ └── No queries
```

**Review by LPTK — APPROVED** [review]
Great, thanks!

**Pull request merged**